chroma_api/stratagem.py

```diff
@@ -0,0 +1,145 @@
+    def is_valid(self, bundle, request=None):
+        purge_duration = bundle.data.get("purge_duration")
+        report_duration = bundle.data.get("report_duration")
+
+        if "filesystem" not in bundle.data:
+            return {"code": "filesystem_required", "message": "filesystem required when running stratagem."}
+        elif isinstance(purge_duration, int) and isinstance(report_duration, int) and report_duration >= purge_duration:
+            return {"code": "duration_order_error", "message": "Report duration must be less than purge duration."}
+
+        fs_identifier = str(bundle.data.get("filesystem"))
+
+        if not any(
+            map(
+                lambda x: str(x.get("id")) == fs_identifier or str(x.get("name")) == fs_identifier,
+                ManagedFilesystem.objects.values("id", "name"),
+            )
+        ):
+            return {
+                "code": "filesystem_does_not_exist",
+                "message": "Filesystem {} does not exist.".format(fs_identifier),
+            }
+
+        fs_id = filter(
+            lambda x: str(x.get("id")) == fs_identifier or str(x.get("name")) == fs_identifier,
+            ManagedFilesystem.objects.values("id", "name"),
+        )[0].get("id")
+
+        # Each MDT associated with the fielsystem must be installed on a server that has the stratagem profile installed
+        target_mount_ids = map(lambda mdt: mdt.active_mount_id, ManagedMdt.objects.filter(filesystem_id=fs_id))
+        host_ids = map(
+            lambda target_mount: target_mount.host_id, ManagedTargetMount.objects.filter(id__in=target_mount_ids)
+        )
+        host_ids = pipe(host_ids, set, list)
+        installed_profiles = map(lambda host: host.server_profile_id, ManagedHost.objects.filter(id__in=host_ids))
+        if not all(map(lambda name: name == "stratagem_server", installed_profiles)):
+            return {
+                "code": "stratagem_server_profile_not_installed",
+                "message": "'stratagem_servers' profile must be installed on all MDT servers.",
+            }
+
+        return {}
+
+
+class StratagemConfigurationValidation(RunStratagemValidation):
+    def is_valid(self, bundle, request=None):
+        required_args = [
+            "interval",
+            "report_duration",
+            "report_duration_active",
+            "purge_duration",
+            "purge_duration_active",
+        ]
+
+        difference = set(required_args) - set(bundle.data.keys())
+
+        if not difference:
+            return super(StratagemConfigurationValidation, self).is_valid(bundle, request)
+
+        return {"__all__": "Required fields are missing: {}".format(", ".join(difference))}
+
+
+class StratagemConfigurationResource(ChromaModelResource):
+    filesystem = fields.CharField(attribute="filesystem_id", null=False)
+    interval = fields.IntegerField(attribute="interval", null=False)
+    report_duration = fields.IntegerField(attribute="report_duration", null=False)
```

Comment:
Can't durations be null?

If they were null, doesn't that mean we don't need the active fields?

Reply:
sure, works for me.